src/apps/sources/pages/ViewSourcePage.tsx

```diff
@@ -1,96 +1,115 @@
-
-    useEffect(() => {
-        retrieveSourceAndDetails(url);
-    }, [url, retrieveSourceAndDetails]);
-
+  profile,
+  usersOrgs = [],
+  sourceLoading,
+  source,
+  retrieveSourceAndDetails,
+  retrieveSourceErrors,
+  retrieveConceptsSummary,
+  metaConceptsCount = {},
+}: Props) => {
+  const { pathname: url, state } = useLocation();
+  const previousPath = state ? state.prevPath : "";
 
+  useEffect(() => {
+    retrieveSourceAndDetails(url);
+  }, [url, retrieveSourceAndDetails]);
+  useEffect(() => {
+    retrieveConceptsSummary(url + "concepts/");
+  }, [url, retrieveConceptsSummary]);
 
-    return (
-        <Header title={`${getSourceTypeFromPreviousPath(previousPath)} > ${source?.name || ""}`} justifyChildren="space-around">
-            <ProgressOverlay
-            delayRender
-            loading={sourceLoading}
-            error={
-                retrieveSourceErrors
-                    ? "Could not load source. Refresh the page to retry"
-                    : undefined
-            }
-        >
-                <Grid id="viewSourcePage" item xs={5} component="div">
-                    <Paper className="fieldsetParent">
-                        <fieldset>
-                            <Typography component="legend" variant="h5" gutterBottom>
-                                General Details
-                            </Typography>
-                            <SourceForm
-                                savedValues={apiSourceToSource(source)}
-                                profile={profile}
-                                usersOrgs={usersOrgs}
-                                loading={true}
-                            />
-                        </fieldset>
-                    </Paper>
-                </Grid>
-            </ProgressOverlay>
-        </Header>
-    );
+  return (
+    <Header
+      title={`${getSourceTypeFromPreviousPath(previousPath)} > ${source?.name ||
+        ""}`}
+      justifyChildren='space-around'
+    >
+      <ProgressOverlay
+        delayRender
+        loading={sourceLoading}
+        error={
+          retrieveSourceErrors
+            ? "Could not load source. Refresh the page to retry"
+            : undefined
+        }
+      >
+        <Grid id='viewSourcePage' item xs={5} component='div'>
+          <Paper className='fieldsetParent'>
+            <fieldset>
+              <Typography component='legend' variant='h5' gutterBottom>
+                General Details
+              </Typography>
+              <SourceForm
+                savedValues={apiSourceToSource(source)}
+                profile={profile}
+                usersOrgs={usersOrgs}
+                loading={true}
+              />
+            </fieldset>
+          </Paper>
+        </Grid>
+        <Grid item xs={5} container spacing={2}>
+          <Grid item xs={12} component='div'>
+            <SourceDetails source={source} totalConceptCount={metaConceptsCount.num_found || 0} />
```

Comment:
@tarunkumar-tw @kirity-mahanty can we have a new test case added for newly added <SourceDetails/> component

Reply:
@binduak We have added test cases to newly added component. Here is the reference for test cases file: src/apps/sources/__test__/components/SourceConceptsSummary.test.tsx